gravitee-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ConfigServiceImpl.java

```diff
@@ -71,9 +71,37 @@ public boolean portalLoginForced() {
 
     @Override
     public PortalConfigEntity getPortalConfig() {
+        return this.getPortalConfig(GraviteeContext.getCurrentEnvironment());
+    }
+
+    @Override
+    public PortalConfigEntity getPortalConfig(String environmentId) {
         PortalConfigEntity portalConfigEntity = new PortalConfigEntity();
         Object[] objects = getObjectArray(portalConfigEntity);
 
+        loadConfigByReference(objects, portalConfigEntity, environmentId, ParameterReferenceType.ENVIRONMENT);
```

Comment:
maybe you can let loadConfigByReference get the object array and avoid this parameter ?